sdk/python/kfp/cli/compile_.py

```diff
@@ -149,7 +157,8 @@ def compile_(
         pipeline_func=pipeline_func,
         pipeline_parameters=parsed_parameters,
         package_path=package_path,
-        type_check=not disable_type_check)
+        type_check=not disable_type_check,
+        enable_caching=enable_caching)
```

Comment:
once you update as per https://github.com/kubeflow/pipelines/pull/11209/files#r1759241709, 
you would need to change this as well to something like:
`enable_caching=not disable_caching_default`